src/core/parser.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <initializer_list>
+#include <nlohmann/json.hpp>
+#include "parse_feeder.hpp"
+#include "parse_worker.hpp"
+#include "comments_parser.hpp"
+#include "ignore_parser.hpp"
+
+namespace docgen {
+namespace core {
+
+class Parser : private ParseWorker
```

Comment:
I also think a `Parser` should own a `ParseWorker`. You're essentially constructing a tree of parse workers. The one that `Parser` owns is the root of that tree. `Parser` sounds more like a manager of resources - a mechanism which you can store the json objects and do stuff with it and the tree of parse workers that will perform parsing.